File: flow-tests/test-root-context/frontend/test-form.js
import {PolymerElement} from '@polymer/polymer/polymer-element.js';
import {html} from '@polymer/polymer/lib/utils/html-tag.js';

class TestForm extends PolymerElement {
  static get is() { return 'test-form' }

  static get template() {
    return html`
       <div id="div">Template text</div> 
    `;
  }
}
customElements.define(TestForm.is, TestForm);

File: flow-tests/test-frontend/vite-production-custom-frontend/src/main/java/com/vaadin/viteapp/littemplate/TestForm.java
package com.vaadin.viteapp.littemplate;

import com.vaadin.flow.component.Tag;
import com.vaadin.flow.component.dependency.JsModule;
import com.vaadin.flow.component.html.Div;
import com.vaadin.flow.component.littemplate.LitTemplate;
import com.vaadin.flow.component.template.Id;

@JsModule("./lit-templates/test-form.js")
@Tag("test-form")
public class TestForm extends LitTemplate {

    @Id
    private Div div;

    public TestForm() {
        div.setText("foo");
    }
}
